thousandeyes/provider.go

```diff
@@ -21,6 +22,11 @@ func Provider() *schema.Provider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("TE_AID", nil),
 			},
+			"timeout": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  schema.EnvDefaultFunc("TE_TIMEOUT", 0),
```

Comment:
will this default to `0`?

Reply:
It will.

In its current state, go-thousandeyes takes one value for the custom timeout, meaning we don't distinguish between 0 and unset.  So it overrides 0 with the same value it uses for default, hence defaulting to 0 here.

The first-thought alternatives to defaulting to 0 here would be to change go-thousandeyes custom timeout to require a separate value to be set or to also set a specific default timeout here.  If we want to change that, I'd favor the former so all users of the client benefit from future changes to the default.

Reply:
@JTBlanchard. So what is the new behavior based on the default you've defined ?

Reply:
The new behavior is to send the timeout specified in the environment variable to go-thousandeyes as now supported: https://github.com/william20111/go-thousandeyes/pull/75/files

If go-thousandeyes library receives a zero for the timeout (as is the default used here when the environment variable is not set), then it uses its internal default timeout.

Reply:
@JTBlanchard can you update the docs to reflect this ?

Reply:
@johntdyer Done, as it should have been from the get-go. Thanks!